src/lean_dojo/data_extraction/traced_data.py

```diff
@@ -158,17 +158,20 @@ def __getstate__(self) -> Dict[str, Any]:
     @property
     def tactic(self) -> str:
         """The raw tactic string."""
+        assert isinstance(self.ast, TacticNode)
```

Comment:
These assertions are only true for Lean 3?

Reply:
Ups, good comment.
I thought it would only be implement for lean_3, as the Node4 does not have the self.ast.tactic. But its more complicated. I will do it in another PR.